src/megaclient.cpp

```diff
@@ -1836,6 +1799,7 @@ void MegaClient::exec()
                             {
                                 syncnaglebt.backoff(nds - Waiter::ds);
                                 syncnagleretry = true;
+                                syncuprequired = true;
```

Comment:
Having a flag per sync instead of a global flag would make the line above unnecessary.